cli/cli.py

```diff
@@ -534,6 +614,29 @@ def get_option_type_info() -> dict[str, Any]:
     return option_type_info
 
 
+def process_tables_options(opts: dict[str, Any]) -> dict[str, Any]:
+    new_tables = []
+    if TABLES_LIST in opts:
+        if not is_omitted(opts[TABLES_LIST]):
+            new_tables.extend(parse_tables_list_file(opts[TABLES_LIST]))
+        opts.pop(TABLES_LIST)
+    if TABLES_DIRECTORY in opts:
+        if not is_omitted(opts[TABLES_DIRECTORY]):
+            new_tables.extend(parse_tables_directory(opts[TABLES_DIRECTORY]))
+        opts.pop(TABLES_DIRECTORY)
+
+    if new_tables:
+        if TABLES in opts:
+            if is_omitted(opts[TABLES]):
+                opts[TABLES] = new_tables
+            else:
+                opts[TABLES].extend(new_tables)
```

Comment:
`click` puts a tuple into the dict, but tuples are immutable, `.extend` won't work